erpnext/hr/doctype/employee_attendance_tool/employee_attendance_tool.py

```diff
@@ -55,8 +55,7 @@ def mark_employee_attendance(employee_list, status, date, leave_type=None, compa
 		else:
 			leave_type = None
 
-		if not company:
-			company = frappe.db.get_value("Employee", employee['employee'], "Company")
+		company = frappe.db.get_value("Employee", employee['employee'], "Company")
```

Comment:
```suggestion
		company = frappe.db.get_value("Employee", employee['employee'], "Company", cache=True)
```

To avoid repeated queries for the same employee.